### ISSUE TITLE
Superheavy Shuttle Bugs Out After Arriving at Destination - Destroying any pawns within

### ISSUE THREAD

**Issue by CyberSavvy531** [SonicTHI/SaveOurShip2Experimental#11]
Hey, guys, I know I've seen mention of shuttles being pretty buggy, but I didn't see mention of it in the official SOS2 sheet. So, I thought I would open this to officially report the issue(s) I've seen. I have SOS2v70 installed, running 1.4 with Royalty, Ideology, and Biotech. I've got a sizable selection of mods, too, but nothing that would immediately cause suspicion for this issue.

I only just started getting this error now. I've been using shuttles up until this point with the only major issue being the inability to draft shuttles in hover mode. But I'll just open a separate issue for that if requested.

I'm loading a superheavy shuttle from a bandit camp quest. Loaded is a mechanitor pawn, 2 tunneller mechs, an incapacitated prisoner, and a selection of other items, including hemogen from Biotech.

Loading and launching the shuttle seems to work without issue. Once the shuttle arrives and lands is when the more critical bug(s) appear. I've attached links to the logs. Save file was too big to upload, but I'll keep a copy in case you guys want it via another way.

Details of what I see in game:

1. When loading the save, the very first error pops up referencing `ShipBlueprintStartShipTachinante`

2. After loading and launching the shuttle and once it lands back at my colony the error referencing `Exception ticking ShuttleIncomingDropship763144` pops up. (There's another error in the logs just before this referring to a different mod. I believe it's related to Pick Up and Haul, and not affecting SOS2.).

3. This is where things get weird. I have the game set to pause the game any time an error is logged. So, immediately after step 2, I can either:

(1) Unpause the game, and it will immediately destroy the mechanitor pawn and the 2 tunneller mechs, **not** the prisoner, though. This throws another error ` there an exception? Destroying skyfaller`. If I then unpause the game and toggle the shuttle out of hover mode the final error pops up: `Tried to destroy already-destroyed thing ShuttleDropshipRace763142`. I can then toggle the shuttle's hover mode repeatedly, and each time I toggle hover mode _off_ that error will pop up again. Those are the other instances you'll see in the log. I did notice that with each cycle of toggling the shuttle's hover mode, the shuttle's number in hover mode is incremented by 1 each time as if I have multiple. So it seems toggling hover mode is creating a new shuttle instance each time and all continue to be tracked even though only one is visually present on the map.

(2) Without unpausing the game, toggle the shuttle's hover mode off. This will drop all cargo, including the prisoner, except for the mechanitor pawn and the 2 tunneller mechs. At this point, I get the `Tried to destroy already-destroyed thing ShuttleDropshipRace763142` error. Once I unpause the game the pawn and mechs are then destroyed, and the other error pops up: `Was there an exception? Destroying skyfaller.`

Option 1 log:  https://gist.github.com/c6e00bc766678d2ffbbc45c43d576b94
Option 2 log:  https://gist.github.com/e2b752f5c6ec278c3c1f478100531190

**Comment by KrzysztofKowalski** [comment]
@CyberSavvy531  i had same result with colonists missing after using sos2 shuttle

**Comment by KrzysztofKowalski** [comment]
workaround for this is to land 1 tile before destination and use normal caravan to enter map 

however still if you pack too many things to shuttle some items outside of circle diameter disappear

**Comment by SonicTHI** [comment]
Can someone test this with V79 and report if its still an issue?

**Comment by CyberSavvy531** [comment]
I can test it later today and update this.

**Comment by CyberSavvy531** [comment]
There were still a couple of errors relating to textures, a null reference exception, and one or two others (log below), but the initial issue of pawns being destroyed seems to be gone. Great job on the updates, guys! It's awesome to see such an awesome modding community.

https://gist.github.com/bb610374fd45a7db86e01ffd45e4e808
